terratorch/cli_tools.py

```diff
@@ -206,30 +214,36 @@ def setup(self, trainer: Trainer, pl_module: LightningModule, stage: str) -> Non
                 # save only on rank zero to avoid race conditions.
                 # the `log_dir` needs to be created as we rely on the logger to do it usually
                 # but it hasn't logged anything at this point
-                fs.makedirs(log_dir, exist_ok=True)
+                if self.deploy_config_file:
+                    fs.makedirs(log_dir, exist_ok=True)
```

Comment:
@Joao-L-S-Almeida  the log_dir is only created if the deploy_config_file is True - is that the intended semantics?